### PR TITLE
Feature/split cluster into small files

### PR DESCRIPTION
**Pull request by 9547**
### What problem does this PR solve? 

ref https://github.com/pingcap/tiup/pull/958#discussion_r538976861

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
NONE
```

### CODE REVIEW

File: components/cluster/command/deploy.go [hunk 1]
```diff
@@ -77,17 +77,7 @@ func newDeploy() *cobra.Command {
...
-				version,
-				topoFile,
-				opt,
-				postDeployHook,
-				skipConfirm,
-				gOpt.OptTimeout,
-				gOpt.SSHTimeout,
-				gOpt.SSHType,
-			)
+			return cm.Deploy(clusterName, version, topoFile, opt, postDeployHook, skipConfirm, gOpt)
```
Comment (lucklove) [inline components/cluster/command/deploy.go:80]: It's better to keep the default manager defined in `manager` package:

```
var clusterManager *clusterManager

func Deploy(...) {
    return clusterManager.Deploy(...)
}
```

Comment (9547) [reply components/cluster/command/deploy.go:80]: I'm sorry, but I'm curious about the advantage of this design pattern.  We used a singleton instance to handle all `Manager` related works, the former declares the instance in user-space, while the latter in the package-space. But the later one seems duplicated, do we need to declare a one-one proxy method for the public methods?

**Pull request merged**